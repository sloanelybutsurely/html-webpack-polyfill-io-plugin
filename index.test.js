'use strict'
const fs = require('fs')
const path = require('path')
const rimraf = require('rimraf')
const webpack = require('webpack')
const HtmlWebpackPlugin = require('html-webpack-plugin')
const HtmlWebpackPolyfillIOPlugin = require('./index')

const OUTPUT_DIR = path.join(__dirname, 'dist')

describe('HtmlWebpackPolyfillIOPlugin', () => {
  beforeEach(done => {
    rimraf(OUTPUT_DIR, done)
  })
  it('adds the polyfill.io script to the html page', done => {
    const compiler = webpack(
      {
        entry: path.join(__dirname, 'fixtures', 'index.js'),
        output: { path: OUTPUT_DIR },
        plugins: [new HtmlWebpackPlugin(), new HtmlWebpackPolyfillIOPlugin()],
      },
      err => {
        expect(err).toBeFalsy()
        const html = fs
          .readFileSync(path.join(OUTPUT_DIR, 'index.html'))
          .toString()
        expect(html).toMatchSnapshot()
        done()
      }
    )
  })
})
