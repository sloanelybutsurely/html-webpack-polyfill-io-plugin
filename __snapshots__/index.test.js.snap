// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HtmlWebpackPolyfillIOPlugin adds the polyfill.io script to the html page 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"UTF-8\\">
    <title>Webpack App</title>
  <script src=\\"https://cdn.polyfill.io/v2/polyfill.min.js\\"></script></head>
  <body>
  <script type=\\"text/javascript\\" src=\\"main.js\\"></script></body>
</html>"
`;
